connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/CustomSparkConnectBlockingStub.scala

```diff
@@ -18,13 +18,93 @@ package org.apache.spark.sql.connect.client
 
 import scala.jdk.CollectionConverters._
 
+import com.google.protobuf.GeneratedMessageV3
 import io.grpc.ManagedChannel
+import io.grpc.stub.StreamObserver
 
 import org.apache.spark.connect.proto._
 
+// This is common logic shared between the blocking and non-blocking stubs.
+//
+// The common logic is responsible to verify the integrity of the response. The invariant is
+// that the same stub instance is used for all requests from the same client.
+private[client] class SparkConnectCommonStub {
```

Comment:
nit: move to it's own file

Also, it's not very different in what it does from RetryHandler, and GrpcExceptionConverter. Why make this one a superclass, while others are members?
Why not make it a member ResponseValidator object and have responseValidator.wrapIterator and responseValidator.validateResponse instead of superclass?

Reply:
Refactored